examples/plot/plot-layer/surface-layer.js

```diff
@@ -199,51 +188,21 @@ export default class SurfaceLayer extends Layer {
       for (let uIndex = 0; uIndex < uCount; uIndex++) {
         const u = uIndex / (uCount - 1);
         const v = vIndex / (vCount - 1);
-        let [x, y, z] = getPosition(u, v);
+        const [x, y, z] = getPosition(u, v);
 
         const isXFinite = isFinite(x);
         const isYFinite = isFinite(y);
         const isZFinite = isFinite(z);
-        if (!isXFinite) {
-          x = 0;
-        }
-        if (!isYFinite) {
-          y = 0;
-        }
-        if (!isZFinite) {
-          z = 0;
-        }
-
-        xMin = Math.min(xMin, x);
-        xMax = Math.max(xMax, x);
-        yMin = Math.min(yMin, y);
-        yMax = Math.max(yMax, y);
-        zMin = Math.min(zMin, z);
-        zMax = Math.max(zMax, z);
 
         // swap z and y: y is up in the default viewport
-        value[i++] = x;
-        value[i++] = z;
-        value[i++] = y;
+        value[i++] = isXFinite ? xScale(x) : xScale.range()[0];
+        value[i++] = isZFinite ? zScale(z) : zScale.range()[0];
+        value[i++] = isYFinite ? yScale(y) : yScale.range()[0];
         value[i++] = isXFinite && isYFinite && isZFinite ? 0 : 1;
       }
     }
 
-    const xScale = getXScale({min: xMin, max: xMax});
-    const yScale = getYScale({min: yMin, max: yMax});
-    const zScale = getZScale({min: zMin, max: zMax});
-
-    for (let j = 0; j < vertexCount; j++) {
-      const startIndex = j * 4;
-      if (!value[startIndex + 3]) {
-        value[startIndex] = xScale(value[startIndex]);
-        value[startIndex + 1] = zScale(value[startIndex + 1]);
-        value[startIndex + 2] = yScale(value[startIndex + 2]);
-      }
-    }
-
     attribute.value = value;
-    this.props.onUpdate({xScale, yScale, zScale});
```

Comment:
Just to make sure I understand this solution correctly, is there a reason that `calculatePositions` in `SurfaceLayer` is not called when the `SurfaceLayer` is initiated, such that `xScale` is not computed and therefore not passed to `PlotLayer` (which results in `AxisLayer` not being rendered)? If that is the case, `x`, `y`, `z` positions in `SurfaceLayer` would not be computed either (and would only be computed until user's interaction). Is that a bug that needs to be fixed as well?

Reply:
`updateState` is called during initiation.

Reply:
I see, then `this.props.onUpdate` should be called too. Why the `AxesLayer` was not shown then? (I was assuming the `AxesLayer` was not shown because its `xScale` etc. was not computed.

Reply:
The `onUpdate` callback is removed in this PR. We used to use this callback to set the state of the parent layer, and then use the new state to draw the axes layer. However, currently there is no way to trigger `CompositeLayer.renderLayers()` from within. Therefore the axes layer was not displayed until the user manipulates the viewport.

@ibgreen I was half expecting `needsRedraw` to serve this purpose. The catch being composite layers' "draw" calls happen in `updateLayers`, while regular layers' draw calls happen in `drawLayers`.

Reply:
I see, thanks for the explanation!

Reply:
> I was half expecting needsRedraw to serve this purpose. The catch being composite layers' "draw" calls happen in updateLayers, while regular layers' draw calls happen in drawLayers.

Yes, I believe we still have some work to do here. I also feel that we don't have a convincing definition of exactly what should trigger composite layer updates / rerenders of sub layers. E.g. `shouldUpdateState` might be implemented differently by different sublayers, but right now we ignore that.